### DISCUSSION TITLE
Modifying the AST and Rewriting the AST to Markdown

### DISCUSSION THREAD

**Discussion by matttproud** [yuin/goldmark#437] [Q&A]
Hi,

goldmark looks like a nice suite of functionality.  I have been examining it, and I am wondering if there is any capability analogous to [`package dst`](https://pkg.go.dev/github.com/dave/dst)'s (a replacement for Go's own [`package ast`](https://pkg.go.dev/go/ast)) ability to dump out modified ASTs with all elements intact ([example](https://pkg.go.dev/github.com/dave/dst#readme-usage)).

Let's say that I wrote my own `parser.ASTTransformer` or `ast.Walker` that I used to modify the Markdown document.  Does goldmark provide a way for me to re-emit the modified AST back to Markdown source (but reflecting the modifications to the AST that I made in the walker or transformer)?

Many thanks in advance!

**Comment by riscdanger** [comment]
Would a text renderer work here? Though I think some information is lost during parsing (such as extra blank lines), so it wouldn't be exact.

Personally, I'd like something lik `go fmt` for markdown to run on pre-commit hooks, rendering all lists with my preferred marker, keeping lines truncated to 80, etc.
